### ISSUE TITLE
Version pinning Gem files

### ISSUE THREAD

**Issue by acsportstec** [SumoLogic/sumologic-collector-chef-cookbook#124]
As the version numbers are not pinned when installing gem files, when using older versions of the `sumologic-collector-chef-cookbook` it will try and install the latest gems.

```
chef_gem 'chef-vault' do
  compile_time true if respond_to?(:compile_time)
end
```

These should be pinned with the current version of the gem to ensure that when running the cookbook on older environments it does not try and force a ruby update:

```
chef_gem 'chef-vault' do
  compile_time true if respond_to?(:compile_time)
  version '2.9.1'
end
```

Error:

```STDERR: ERROR:  Error installing chef-vault:

    chef-vault requires Ruby version >= 2.2.0.
---- End output of C:/opscode/chef/embedded/bin/gem install chef-vault -q --no-rdoc --no-ri -v "3.0.1" ----
```

**Comment by dhui** [comment]
Fixed by my PR: https://github.com/SumoLogic/sumologic-collector-chef-cookbook/pull/123

**Comment by majormoses** [comment]
closed via #123
